internal/indexmanagement/reconcile.go

```diff
@@ -379,7 +381,12 @@ func (imr *IndexManagementRequest) reconcileIndexManagementCronjob(policy apis.I
 
 	// prune-namespaces cron job
 	if policy.Phases.Delete != nil && policy.Phases.Delete.Namespaces != nil {
-		schedule, err := crontabScheduleFor(policy.Phases.Delete.PruneNamespacesInterval)
+		pruneNamespaceIntervalTime := pruneNamespaceInterval
+		if policy.Phases.Delete.PruneNamespacesInterval != "" {
+			pruneNamespaceIntervalTime = policy.Phases.Delete.PruneNamespacesInterval
+		}
```

Comment:
Should we not just make the interval mandatory for executing the jobs and just debug out a message versus substituting a default value?

Reply:
TBH this is not important thing from a customer point of view and making it mandatory doesn't make sense to me. It doesn't effect a cronjob's intended task in any way.
Also, I would like to keep it optional for consistency with the previous pollInterval for normal cronjobs.

Reply:
Maybe it's more of a question of which side should reinforce the mandate for the interval?

Reply:
What do you say about it? Which side you feel should we mandate?

Reply:
For consistency, I would think the CLO api as it does that for other policies.

Reply:
You mean make it optional in EO and madate to set it through CLO right? If yes, I am doing that.

Reply:
The constant here will help for testing if we directly create a CR from EO. Which making changes to CLO apis I have made this field mandatory.

Reply:
Yes... should be optional in EO cause it's a new feature. I guess what I'm seeing is that having both `Namespaces` and `Interval` be mandatory for executing the policy. Since CLO already puts them in here, it won't matter if we block that action in EO.

So, pretty much I don't have a problem with the conditionals. My thought is mainly should EO be adding the default or just skipping it as bad configuration.

Reply:
Some people test it only by deploying EO and hence I thought to put this check and a default value. I can remove it if you feel it's a bad configuration?

Reply:
The only reason I would call it bad configuration is because we already assume that the `PollInterval` exists for the other jobs, so I would assume that someone would define it here. Maybe we can just add a note that it needs both and remove the default value?

Reply:
Updated @Red-GV